src/Messenger/DataImporterHandler.php

```diff
@@ -50,8 +50,13 @@ public function __construct(
 
     public function __invoke(DataImporterMessage $message)
     {
-        foreach ($message->getIds() as $id) {
+        foreach ($message->getIds() as $i => $id) {
             $this->importProcessingService->processQueueItem($id);
+
+            // call the garbage collector to avoid too many connections & memory issue
+            if (($i + 1) % 200 === 0) {
```

Comment:
not needed, as worker item count is 200 by default anyway: https://github.com/pimcore/data-importer/blob/1.x/src/DependencyInjection/Configuration.php#L53

so, by default in one message, there should not be more than 200 items anyway.

Reply:
what about the cases where `worker_item_count` is set higher in config?

Reply:
if this leads to problems, the developer needs to reduce it again :-)
the whole point of that worker_item_count is to prevent these kind of problems... memory, long running processes, etc.

Reply:
Reverted. thanks!